File: habitat.cpp
#include "habitat.h"
#include <iostream>
using namespace std;
Habitat::Habitat():Cell(){};
char* Habitat::GetName()
{
  char* c=(char*)"habitat";
  return c;
}
LandHabitat::LandHabitat():Habitat(){};
char* LandHabitat::GetType()
{
  char* c=(char*)"LandHabitat";
  return c;
}
void LandHabitat::Render()
{
  cout<<"l";
}


WaterHabitat::WaterHabitat():Habitat(){};
char* WaterHabitat::GetType()
{
  char* c=(char*)"WaterHabitat";
  return c;
}
void WaterHabitat::Render()
{
  cout<<"w";
}

AirHabitat::AirHabitat():Habitat(){};
char* AirHabitat::GetType()
{
  char* c=(char*)"AirHabitat";
  return c;
}
void AirHabitat::Render()
{
  cout<<"a";
}

File: Habitat.java
/**
 * @file	habitat.h
 * @author	Harum Lokawati
 * @date	March 2017
 * @version VZ03
 * 
 * @brief	the header file containing class declaration Habitat
 */

/**
 * @class	Habitat
 * @brief	acstract class of Habitat as one of Cell
 */
abstract class Habitat extends Cell{
    /**
	 * @brief default construstor
	 * this will construct Cell
	 */
    public Habitat();
        /**
	 * @brief name getter
	 * this will return "habitat"
         * @return char*
		 */
    public String GetName();
        /**
	 * @brief pure virtual function for type getter
	 */
    abstract String GetType();
};
